### ISSUE TITLE
FileBotTool not activateable in plugin screen

### ISSUE THREAD

**Issue by geldofa** [Laharah/deluge-FileBotTool#53]
Hi,

I have been trying to get the FileBottool to work with Deluge 2.1.1 but can't seem to get it right.

Tried a normal plugin install, manually putting it in the folder, distilling the 2.0 & 2.0.2 versions in py3.8 & py3.9, editing core.conf but with no success. 

Is there something als i could try? Think i have exhausted the available info on this online.

Deluge: 2.1.1
Filebot: 4.9.6
OS: Windows 10

Thx in advance for the replies

**Comment by Laharah** [comment]
Can you check and make sure that the `.egg` filename ends with the same python version that your deluge runs? (you can get this info on the `about` page in deluge.) 

For example if deluge is running on python 3.8, make sure that the `.egg` filename is `FileBotTool-2.0.2-py3.8.egg`.

**Comment by geldofa** [comment]
Hey Laharah, thx for your reply!

AppData\Roaming plugin folder: i can activate py3.8 eggs which work without a problem.
Program Files plugin folder: py3.9 eggs for the standard plugins.

I also tried a clean reinstall (I upgraded previously) and as first order installing FileBotTool with no success.
Could there we a conflict with me having 2 versions of python installed on my system? (I was able to distil py3.8 & py3.9 eggs)

As for permission, if i look at access my account has the same permissions as an administrator

Sidenote: Thanks for this wonderfull tool, i am already using it for a couple of years now and i really love what you have made.

**Comment by Laharah** [comment]
It sounds like your deluge is running on python 3.8, if you go into the plugin folder and rename the plugin to `FileBotTool-2.0.2-py3.8.egg` it may work, at least it has for people in the past.

**Comment by Laharah** [comment]
Hmm, any way you can get some debug logs to me? I'm not sure what else could be causing it.

**Comment by fatlewis** [comment]
I'm having an issue that may be identical to this; Deluge 2.1.1, FileBot 4.9.6, Win10, plugin is displayed in the list but fails to activate.

These are the logs I'm getting when I click to activate the plugin:

```
16:33:39 [ERROR   ][deluge.pluginmanagerbase          :72  ] Unable to instantiate plugin 'FileBotTool' from 'c:\\users\\noget\\appdata\\roaming\\deluge\\plugins\\filebottool-2.0.2-py3.9.egg'!
16:33:39 [ERROR   ][deluge.pluginmanagerbase          :72  ] No module named 'winprocess'
Traceback (most recent call last):
  File "deluge\pluginmanagerbase.py", line 148, in enable_plugin
  File "c:\users\noget\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\__init__.py", line 47, in __init__
    from .core import Core as _plugin_cls
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\noget\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\core.py", line 57, in <module>
    from . import pyfilebot
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\noget\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\pyfilebot.py", line 19, in <module>
    from . import killableprocess
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\noget\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\killableprocess.py", line 69, in <module>
    import winprocess
ModuleNotFoundError: No module named 'winprocess'
```

Unless I've screwed something up with my python install it seems to be some kind of module import issue?

**Comment by Laharah** [comment]
@fatlewis, that's weird. There is a winprocess module included in the .egg file. 

If you're able to, you can try changing line 69 in `killableprocess.py` to `from . import winprocess`. If you then follow the [build instructions](https://github.com/Laharah/deluge-FileBotTool#how-to-build) on the readme, you can see if the new .egg file works.

**Comment by GentleHoneyLover** [comment]
I have the same issue when trying to add plugin to deluge 2.0.3 — it simply doesn't add.

Tried on Ubuntu 20.04 (with python 3.8) and on Ubuntu 22.04 (with python 3.10). Was renaming the .egg file like suggested. No luck.

Any ideas? Should I try building it from source?

**Comment by Laharah** [comment]
@GentleHoneyLover If you can get me logs of when you try to add and enable FileBotTool I can get a better idea of what's going on. I don't think it's the same problem as @fatlewis, because his codepath is only activated on windows.

**Comment by GentleHoneyLover** [comment]
@Laharah, unfortunately, deluge doesn't write anything in the log when I attempt to add the plugin for some reason.

I even tried compiling it (following your instructions) — which seems to have been successful. But my deluge still doesn't want to take this newly compiled .egg file.

After I attempt to add the plugin, the following zip file appears in the `plugins` folder of deluge's config folder: `[object FileList]` (no extension). The contents of this zip file is the following:
```
.
├── EGG-INFO
│   ├── dependency_links.txt
│   ├── entry_points.txt
│   ├── not-zip-safe
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   └── top_level.txt
└── filebottool
    ├── __init__.py
    ├── __pycache__
    │   ├── __init__.cpython-310.pyc
    │   ├── auto_sort.cpython-310.pyc
    │   ├── common.cpython-310.pyc
    │   ├── core.cpython-310.pyc
    │   ├── events.cpython-310.pyc
    │   ├── killableprocess.cpython-310.pyc
    │   ├── pyfilebot.cpython-310.pyc
    │   ├── six.cpython-310.pyc
    │   ├── winprocess.cpython-310.pyc
    │   └── wintypes.cpython-310.pyc
    ├── auto_sort.py
    ├── common.py
    ├── core.py
    ├── data
    │   ├── config.glade
    │   ├── config.ui
    │   ├── fb_icon16.png
    │   ├── fb_icon24.png
    │   ├── handler_editor.glade
    │   ├── handler_editor.ui
    │   ├── rename.glade
    │   └── rename.ui
    ├── events.py
    ├── gtkui
    │   ├── __init__.py
    │   ├── __pycache__
    │   │   ├── __init__.cpython-310.pyc
    │   │   ├── common.cpython-310.pyc
    │   │   ├── common_gtk3.cpython-310.pyc
    │   │   ├── config_ui.cpython-310.pyc
    │   │   ├── config_ui_gtk3.cpython-310.pyc
    │   │   ├── gtk3ui.cpython-310.pyc
    │   │   ├── gtkui.cpython-310.pyc
    │   │   ├── handler_editor.cpython-310.pyc
    │   │   ├── handler_editor_gtk3.cpython-310.pyc
    │   │   ├── handler_ui.cpython-310.pyc
    │   │   ├── handler_ui_gtk3.cpython-310.pyc
    │   │   ├── rename_dialog.cpython-310.pyc
    │   │   ├── rename_dialog_gtk3.cpython-310.pyc
    │   │   ├── user_messenger.cpython-310.pyc
    │   │   └── user_messenger_gtk3.cpython-310.pyc
    │   ├── common.py
    │   ├── common_gtk3.py
    │   ├── config_ui.py
    │   ├── config_ui_gtk3.py
    │   ├── gtk3ui.py
    │   ├── gtkui.py
    │   ├── handler_editor.py
    │   ├── handler_editor_gtk3.py
    │   ├── handler_ui.py
    │   ├── handler_ui_gtk3.py
    │   ├── rename_dialog.py
    │   ├── rename_dialog_gtk3.py
    │   ├── user_messenger.py
    │   └── user_messenger_gtk3.py
    ├── killableprocess.py
    ├── pyfilebot.py
    ├── six.py
    ├── winprocess.py
    └── wintypes.py
```


I'm running deluge 1.3 and filebot 4.9.6 in a container (here's my image: [link](https://github.com/GentleHoneyLover/deluge-filebot-docker)). I have multiple plugins installed (including filebot plugin) and everything is working flawlessly. Now, I'm upgrading the image base from ubuntu 18.04 to ubuntu 22.04, which automatically gives me deluge 2.0.3. I'm running this new test container in Docker Desktop on my M1 Mac. Maybe there's something in my image that's causing the problem with plugin installation. I will try another image and report back.

**Comment by GentleHoneyLover** [comment]
@Laharah, ok, I just tested on the [linuxserver.io](https://github.com/linuxserver/docker-deluge) image and the behavior is identical to my container (python version matches in both cases — 3.10.8. Nothing gets written in the log. The same archive file appears in the `plugins` folder.

**Comment by Laharah** [comment]
@GentleHoneyLover, how did you get filebot installed on the linuxserver.io image? I don't think it comes with java, what steps did you do to install it?

I did get the image running and I did add filebottool to deluge by placing it into the plugins folder. I had to change to filename to `FileBotTool-2.0.2-py3.10.egg`, but once I did that I started seeing filebottool errors in deluge about not being able to find a filebot install.

**Comment by Soitora** [comment]
> @fatlewis, that's weird. There is a winprocess module included in the .egg file.
> 
> If you're able to, you can try changing line 69 in `killableprocess.py` to `from . import winprocess`. If you then follow the [build instructions](https://github.com/Laharah/deluge-FileBotTool#how-to-build) on the readme, you can see if the new .egg file works.

My error is the same, `Deluge 2.1.1` on `Windows 10` with `Python 3.9`:

```py
15:34:30 [ERROR   ][deluge.pluginmanagerbase          :72  ] Unable to instantiate plugin 'FileBotTool' from 'c:\\users\\miyura\\appdata\\roaming\\deluge\\plugins\\filebottool-2.0.2-py3.9.egg'!
15:34:30 [ERROR   ][deluge.pluginmanagerbase          :72  ] No module named 'winprocess'
Traceback (most recent call last):
  File "deluge\pluginmanagerbase.py", line 148, in enable_plugin
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\__init__.py", line 47, in __init__
    from .core import Core as _plugin_cls
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\core.py", line 57, in <module>
    from . import pyfilebot
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\pyfilebot.py", line 19, in <module>
    from . import killableprocess
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.2-py3.9.egg\filebottool\killableprocess.py", line 69, in <module>
    import winprocess
ModuleNotFoundError: No module named 'winprocess'
```

And if I do what you suggested (and also bump to 2.0.3):
![image](https://user-images.githubusercontent.com/10836780/209968405-3cb97375-ba13-4dc2-899b-d5b4a6b74a95.png)

```
15:45:03 [ERROR   ][deluge.pluginmanagerbase          :72  ] Unable to instantiate plugin 'FileBotTool' from 'c:\\users\\miyura\\appdata\\roaming\\deluge\\plugins\\filebottool-2.0.3-py3.9.egg'!
15:45:03 [ERROR   ][deluge.pluginmanagerbase          :72  ] No module named 'wintypes'
Traceback (most recent call last):
  File "deluge\pluginmanagerbase.py", line 148, in enable_plugin
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\__init__.py", line 47, in __init__
    from .core import Core as _plugin_cls
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\core.py", line 57, in <module>
    from . import pyfilebot
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\pyfilebot.py", line 19, in <module>
    from . import killableprocess
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\killableprocess.py", line 69, in <module>
    from . import winprocess
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\winprocess.py", line 27, in <module>
    from wintypes import BOOL, BYTE, DWORD, HANDLE, LPCWSTR, LPWSTR, UINT, WORD
ModuleNotFoundError: No module named 'wintypes'
```

Doing the same with `wintypes` just threw a whole new error:
![image](https://user-images.githubusercontent.com/10836780/209969671-9a319912-5050-4a3c-b618-f7adec3b36bc.png)

```py
15:48:32 [ERROR   ][deluge.pluginmanagerbase          :72  ] Error -5 while decompressing data: incomplete or truncated stream
Traceback (most recent call last):
  File "deluge\pluginmanagerbase.py", line 148, in enable_plugin
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\__init__.py", line 47, in __init__
    from .core import Core as _plugin_cls
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\core.py", line 57, in <module>
    from . import pyfilebot
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\pyfilebot.py", line 19, in <module>
    from . import killableprocess
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 664, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 627, in _load_backward_compatible
  File "<frozen zipimport>", line 259, in load_module
  File "c:\users\miyura\appdata\roaming\deluge\plugins\filebottool-2.0.3-py3.9.egg\filebottool\killableprocess.py", line 69, in <module>
    from . import winprocess
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 982, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 925, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1423, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1397, in _get_spec
  File "<frozen importlib._bootstrap_external>", line 1378, in _legacy_get_spec
  File "<frozen importlib._bootstrap>", line 423, in spec_from_loader
  File "<frozen importlib._bootstrap_external>", line 716, in spec_from_file_location
  File "<frozen zipimport>", line 191, in get_filename
  File "<frozen zipimport>", line 709, in _get_module_code
  File "<frozen zipimport>", line 571, in _get_data
zlib.error: Error -5 while decompressing data: incomplete or truncated stream
```

**Comment by viseniv** [comment]
I have the same problem, I can't activate the plugin (python version matches the name)

**Comment by Laharah** [comment]
I was able to finally re-create and (hopefully) fix this bug on a fresh windows VM. The [new release 2.0.3](https://github.com/Laharah/deluge-FileBotTool/releases/tag/2.0.3) should fix this issue. If someone here can confirm or send logs of new errors I can either close this issue or do some more bug hunting.

**Comment by RonSijm** [comment]
> I was able to finally re-create and (hopefully) fix this bug on a fresh windows VM. The [new release 2.0.3](https://github.com/Laharah/deluge-FileBotTool/releases/tag/2.0.3) should fix this issue. If someone here can confirm or send logs of new errors I can either close this issue or do some more bug hunting.

I had this issue in my watchlist and I was waiting for a fix, with the previous version whenever I tried to enable FileBotTool it would instantly disable again.

With this [new release 2.0.3](https://github.com/Laharah/deluge-FileBotTool/releases/tag/2.0.3) - When I enabled it it stayed enabled, a CMD popup showed up, and the FileBotTool menu item is there 😄 - So I can confirm it appears to be working again.

I still have to re-add my sorting rules and see if it actually functions as well, but so far, so good. Thanks for the fix!
